src/rpp/rpp/sources/defer.hpp

```diff
@@ -0,0 +1,50 @@
+//                   ReactivePlusPlus library
+//
+//           Copyright Aleksey Loginov 2023 - present.
+//  Distributed under the Boost Software License, Version 1.0.
+//     (See accompanying file LICENSE_1_0.txt or copy at
+//           https://www.boost.org/LICENSE_1_0.txt)
+//
+//  Project home: https://github.com/victimsnino/ReactivePlusPlus
+
+#pragma once
+
+#include <rpp/sources/fwd.hpp>
+#include <rpp/observables/observable.hpp>
+
+namespace rpp::details
+{
+template<constraint::decayed_type Type, typename Factory>
```

Comment:
Let's remove `Type` . It is deductible by factory

Reply:
Hmm but this will invoke `std::invoke_result_t` two times and this might hurt compilation time a bit no?

Reply:
Maybe, but i don't think a lot. On other hand, it would not add template parameter and would not keep it everywhere while it is easy deducible

Reply:
done